bugsnag-android-core/src/main/java/com/bugsnag/android/Journal.java

```diff
@@ -0,0 +1,170 @@
+     *
+     * @param bytes The bytes to deserialize from
+     * @return The journal
+     */
+    public static @NonNull Journal deserialize(@NonNull byte[] bytes, @NonNull Logger logger) {
+        Journal journal = new Journal();
+        int currentStart = 0;
+        for (int i = 0; i < bytes.length; i++) {
+            if (bytes[i] == 0) {
+                try {
+                    byte[] document = Arrays.copyOfRange(bytes, currentStart, i);
+                    journal.add(Command.deserialize(document));
+                    currentStart = i + 1;
+                } catch (Exception exception) {
+                    logger.e(
+                            "Could not deserialize journal entry" + Arrays.toString(bytes),
+                            exception);
+                }
+            }
+        }
+        return journal;
+    }
+
+    /**
+     * Represents a single journal command, consisting of a document path,
+     * and a value to be applied at that path level.
+     */
+    public static class Command {
+        private static final DslJson<Object> dslJson = new DslJson<>();
+
+        @NonNull
+        public final DocumentPath path;
+        @NonNull
+        public final String pathAsString;
+        @Nullable
+        public final Object value;
+
+        /**
+         * Constructor
+         *
+         * @param path  The document path to modify.
+         * @param value The modification to perform.
+         */
+        public Command(@NonNull String path, @Nullable Object value) {
+            this.pathAsString = path;
+            this.value = value;
+            this.path = new DocumentPath(path);
+        }
+
+        /**
+         * Apply this journal command to a document.
+         *
+         * @param document The document to modify.
+         * @return The resulting document (may not be the same as the one passed in).
+         */
+        public @NonNull Object apply(@NonNull Object document) {
+            return path.modifyDocument(document, value);
+        }
+
+        /**
+         * Serialize to an OutputStream
+         *
+         * @param out The stream
+         * @throws IOException If an IO exception occurs
+         */
+        public void serialize(@NonNull OutputStream out) throws IOException {
+            Map<String, Object> command = new HashMap<>();
+            command.put(this.pathAsString, this.value);
+            dslJson.serialize(command, out);
+        }
+
+        /**
+         * Deserialize from a byte buffer.
+         *
+         * @param document The document to deserialize from
+         * @return The journal command
+         * @throws IOException If an IO exception occurs
+         */
+        @SuppressWarnings("unchecked")
+        public static @NonNull
+        Command deserialize(@NonNull byte[] document)
+                throws IOException {
+            Map<String, Object> map = dslJson.deserialize(Map.class, document, document.length);
```

Comment:
```suggestion
            @SuppressWarnings("unchecked") Map<String, Object> map = dslJson.deserialize(Map.class, document, document.length);
```

Keeping this suppression local will avoid accidentally suppressing any problems that might get introduced in future.